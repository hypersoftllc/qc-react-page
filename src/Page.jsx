import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function Page(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`.trim(),
  }
  delete domProps.children
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

Page.defaultProps = {
  className: '',
  compClassName: 'Page',
  compType: 'div',
}

Page.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf(['div', 'section']),
}

export default Page
