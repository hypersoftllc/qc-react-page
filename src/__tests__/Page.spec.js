/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import Page from '../Page'


describe('Page', () => {

  it('should be a function', () => {
    expect(typeof Page).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <Page/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Page/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Page>
            <span>Child</span>
          </Page>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Page className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Page compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Page className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Page className="foo" compClassName="page"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compType', () => {

      it('should render with expected component', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Page compType="section"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Page compType="div"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
