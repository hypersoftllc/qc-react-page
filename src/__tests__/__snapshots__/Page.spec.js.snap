// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page rendered with children should render children 1`] = `
<div
  className="Block Page"
>
  <span>
    Child
  </span>
</div>
`;

exports[`Page rendered with className should render with expected class attribute 1`] = `
<div
  className="Block Page Foo"
/>
`;

exports[`Page rendered with compClassName should render with expected class attribute 1`] = `
<div
  className="Block"
/>
`;

exports[`Page rendered with compClassName should render with expected class attribute 2`] = `
<div
  className="Block foo"
/>
`;

exports[`Page rendered with compClassName should render with expected class attribute 3`] = `
<div
  className="Block page foo"
/>
`;

exports[`Page rendered with compType should render with expected component 1`] = `
<section
  className="Block Page"
/>
`;

exports[`Page rendered with compType should render with expected component 2`] = `
<div
  className="Block Page"
/>
`;

exports[`Page rendered with no props should use default props 1`] = `
<div
  className="Block Page"
/>
`;
